src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,195 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+public class AuditConfig {
+    private static final List<String> DEFAULT_IGNORED_USERS = Collections.singletonList("kibanaserver");
+    private static final List<String> DEFAULT_DISABLED_CATEGORIES =
+            Arrays.asList(AuditCategory.AUTHENTICATED.toString(),
+                    AuditCategory.GRANTED_PRIVILEGES.toString());
+
+    private final boolean restAuditingEnabled;
+    private final boolean transportAuditingEnabled;
+    private final boolean resolveBulkRequests;
+    private final boolean logRequestBody;
+    private final boolean resolveIndices;
+    private final boolean excludeSensitiveHeaders;
+    private final List<String> ignoredAuditUsers;
```

Comment:
Ignore users accepts pattern which may cause an issue with ordering and pattern matching.

Reply:
If an order is important (I don't think it is, as it is not "match first" or "match last"), use `Set` that preserves order (`LinkedHashSet`) as the implementation.

Reply:
This would propagate passing a higher level structure LinkedHashSet. We can't simply use Set type because it ll be important for consumers to know the underlying implementation that the set is ordered.

Reply:
I looked into the current usage and don't think ordering matters so I ll move to Set with HashSet implementation